staging/src/k8s.io/kubectl/pkg/cmd/cmd_test.go

```diff
@@ -132,22 +133,40 @@ func TestKubectlSubcommandShadowPlugin(t *testing.T) {
 			t.Run(test.name, func(t *testing.T) {
 				pluginsHandler := &testPluginHandler{
 					pluginsDirectory: "plugin/testdata",
+					validPrefixes:    plugin.ValidPluginFilenamePrefixes,
 				}
 				ioStreams, _, _, _ := genericiooptions.NewTestIOStreams()
-
 				root := NewDefaultKubectlCommandWithArgs(KubectlOptions{PluginHandler: pluginsHandler, Arguments: test.args, IOStreams: ioStreams})
-				if err := root.Execute(); err != nil {
-					t.Fatalf("unexpected error: %v", err)
+				if pluginsHandler.lookedup || pluginsHandler.executed {
```

Comment:
Can we simplify this to?;

```go
if !pluginsHandler.lookedup && !pluginsHandler.executed {
        root.SetArgs(test.args[1:])
        if err := root.Execute(); err != nil {
              t.Fatalf("unexpected error: %v", err)
        }
}
```

Reply:
I can change it if you prefer, but I think it better shows the intent of the test if we make the if/else more verbose. Especially when there are multiple comments for each section, so I would opt for better readability in the test code.

Reply:
I'm fine adding comments, especially
```
// args must be set, otherwise Execute will use os.Args (args used for starting the test) and test.args would not be passed
// to the command which might invoke only "kubectl" without any additional args and give false positives
```
seems pretty useful to explain why we are passing `test.args[1:]` explicitly. 

But considering that the other comments can also be extracted by the person who work on unit tests by debugging the code and I'd not prefer sacrificing conventions(no code, only comment between if - else).

Reply:
Sounds good, thanks